widgets/src/main/java/org/pentaho/mantle/client/dialogs/folderchooser/SelectFolderDialog.java

```
@@ -119,15 +121,32 @@ public void onFailure( Throwable caught ) {
     content.add( treeWrapper );
 
     setContent( content );
+    fetchRepository( getSelectedPath() );
+
     TreeItem selItem = tree.getSelectedItem();
     if ( selItem != null ) {
       DOM.scrollIntoView( selItem.getElement() );
     }
   }
 
   public String getSelectedPath() {
-    String selectedPath = ( (FolderTreeItem) tree.getSelectedItem() ).getRepositoryFile().getPath();
-    return selectedPath;
+    final FolderTreeItem selectedItem = (FolderTreeItem) tree.getSelectedItem();
+    return selectedItem != null ? selectedItem.getRepositoryFile().getPath() : getDefaultSaveLocation();
   }
 
+  private void fetchRepository( final String selectedPath ) {
+
+    tree.fetchRepositoryFileTree( new AsyncCallback<RepositoryFileTree>() {
+
+      @Override
+      public void onSuccess( RepositoryFileTree result ) {
+        tree.select( selectedPath );
+      }
+
+      @Override
+      public void onFailure( Throwable caught ) {
+        // do nothing
```

Shouldn't just swallow the failure I'd think. If nothing else, log the message.